configure.ac

```diff
@@ -3747,6 +3747,14 @@ AS_IF([test x"$MJIT_SUPPORT" = "xyes"],
 
 AC_SUBST(MJIT_SUPPORT)
 
+AC_ARG_ENABLE(compile-commands,
+        AS_HELP_STRING([--enable-compile-commands],
+        [enable compile commands[default=no]]),
+        [COMPILE_COMMANDS=$enableval], [COMPILE_COMMANDS=no])
+
+AS_IF([test x"$COMPILE_COMMANDS" = "xyes"],
+	[RUBY_APPEND_OPTIONS(CPPFLAGS, "-MJ \$@.json")])
```

Comment:
Adding to `CPPFLAGS` means all extensions will be affected too.
Intentional?

Reply:
No.  I'd rather not impact extensions, but I don't know how to modify _only_ the Ruby build.  Is there a better variable to use?

Reply:
`XCFLAGS` is used only for the Ruby itself.

Reply:
`XCFLAGS` was unexpectedly extended earlier.
Fixed at 9205b8c8010f474b2d301ccc22335c9a2c4a6cae.